core/src/main/java/org/apache/iceberg/avro/AvroSchemaUtil.java

```diff
@@ -138,6 +138,22 @@ public static boolean isOptionSchema(Schema schema) {
     return false;
   }
 
+  /**
+   * This method decides whether a schema is single type union
+   *
+   * single type union: a schema is of type union and there is only one option
+   *
+   * @param schema input schema
+   * @return true if schema is single type union
+   */
+  public static boolean isSingleTypeUnion(Schema schema) {
+    if (schema.getType() == UNION && schema.getTypes().size() == 1) {
+      return true;
+    }
+
+    return false;
```

Comment:
We can simplify to just:
```suggestion
    return schema.getType() == UNION && schema.getTypes().size() == 1;
```

Reply:
Thanks for the suggestions. Updated the method.